packages/tools/src/utilities/contours/interpolation/findInterpolationList.ts

```diff
@@ -0,0 +1,216 @@
+import getInterpolationData from './getInterpolationData';
+import type { InterpolationViewportData } from '../../../types/InterpolationTypes';
+
+/**
+ * findInterpolationList - Finds the list of contours to interpolate,
+ * including whether they are new contours, or auto generated contours that need
+ * to be updated.
+ *
+ * @param toolData - object, The tool data of the roi contour.
+ * @returns object, An object containing the ROIContourData and the
+ * interpolationList.
+ */
+
+function generateInterpolationData(
+  toolData,
+  viewportData: InterpolationViewportData
+) {
+  console.log(
+    'generateInterpolationData on key interpolationUID',
+    viewportData.interpolationUID
+  );
+  const interpolationData = getInterpolationData(viewportData, [
+    {
+      key: 'interpolationUID',
+      value: viewportData.interpolationUID,
+    },
+  ]);
+  const extent = _getExtentOfRegion(interpolationData);
+  const sliceEdited = _getSlicePositionOfToolData(
+    interpolationData,
+    toolData.annotationUID
+  );
+  const interpolationList = [];
+
+  // Check if contours between the extent can be interpolated.
+  for (let i = extent[0] + 1; i <= extent[1] - 1; i++) {
+    if (_sliceNeedsInterpolating(interpolationData, i)) {
+      console.log('Slice needs interpolating', i);
+      const contourPair = _getBoundingPair(i, extent, interpolationData);
+
+      if (
+        contourPair &&
+        (contourPair[0] === sliceEdited || contourPair[1] === sliceEdited)
+      ) {
+        _appendInterpolationList(contourPair, interpolationList, i);
+      }
+    }
+  }
+
+  return {
+    interpolationData,
+    interpolationList,
+  };
+}
+
+/**
+ * _getExtentOfRegion - Returns a 2 element array with the slice locations of
+ * top and bottom polygon of the ROIContour.
+ *
+ * @param ROIContourData - object, Data on the slice location of contours
+ *                                  for the ROIContour.
+ * @returns Number[], The slice locations of the top and bottom polygon of the ROIContour.
+ */
+
+function _getExtentOfRegion(interpolationData) {
+  const extent = [];
+
+  for (let i = 0; i < interpolationData.length; i++) {
+    if (interpolationData[i].annotations) {
+      extent.push(i);
+      break;
+    }
+  }
+
+  for (let i = interpolationData.length - 1; i >= 0; i--) {
+    if (interpolationData[i].annotations) {
+      extent.push(i);
+      break;
+    }
+  }
+
+  return extent;
+}
+
+/**
+ * _getSlicePositionOfToolData - Finds the slice that was edited.
+ *
+ * @param ROIContourData - description
+ * @param polygonUid - description
+ * @returns description
+ */
+function _getSlicePositionOfToolData(interpolationData, annotationUID) {
+  for (let i = 0; i < interpolationData.length; i++) {
+    if (interpolationData[i].annotations) {
```

Comment:
can we do something like 

const sliceAnnotations = interpolationData[i].annotations

sicne you use it twice